src/EditorFeatures/CSharp/Highlighting/KeywordHighlighters/AsyncAwaitHighlighter.cs

```diff
@@ -26,93 +33,106 @@ protected override bool IsHighlightableNode(SyntaxNode node)
-                        }
-                        break;
-                    }
+                    TryAddAsyncModifier(methodDeclaration.Modifiers, spans);
+                    return;
                 case LocalFunctionStatementSyntax localFunction:
-                    {
-                        var asyncModifier = localFunction.Modifiers.FirstOrDefault(m => m.Kind() == SyntaxKind.AsyncKeyword);
-                        if (asyncModifier.Kind() != SyntaxKind.None)
-                        {
-                            spans.Add(asyncModifier.Span);
-                        }
-                        break;
-                    }
+                    TryAddAsyncModifier(localFunction.Modifiers, spans);
+                    return;
                 case AnonymousFunctionExpressionSyntax anonymousFunction:
-                    if (anonymousFunction.AsyncKeyword.Kind() == SyntaxKind.AsyncKeyword)
-                    {
-                        spans.Add(anonymousFunction.AsyncKeyword.Span);
-                    }
-                    break;
-
-                case AwaitExpressionSyntax awaitExpression:
-                    // Note if there is already a highlight for the previous token, merge it
-                    // with this span. That way, we highlight nested awaits with a single span.
-                    var handled = false;
-                    var awaitToken = awaitExpression.AwaitKeyword;
-                    var previousToken = awaitToken.GetPreviousToken();
-                    if (!previousToken.Span.IsEmpty)
-                    {
-                        var index = spans.FindIndex(s => s.Contains(previousToken.Span));
-                        if (index >= 0)
-                        {
-                            var span = spans[index];
-                            spans[index] = TextSpan.FromBounds(span.Start, awaitToken.Span.End);
-                            handled = true;
-                        }
-                    }
-
-                    if (!handled)
-                    {
-                        spans.Add(awaitToken.Span);
-                    }
-                    break;
-
+                    TryAddAsyncOrAwaitKeyword(anonymousFunction.AsyncKeyword, spans);
+                    return;
                 case UsingStatementSyntax usingStatement:
-                    if (usingStatement.AwaitKeyword.Kind() == SyntaxKind.AwaitKeyword)
-                    {
-                        spans.Add(usingStatement.AwaitKeyword.Span);
-                    }
-                    break;
-
+                    TryAddAsyncOrAwaitKeyword(usingStatement.AwaitKeyword, spans);
+                    return;
                 case LocalDeclarationStatementSyntax localDeclaration:
-                    if (localDeclaration.AwaitKeyword.Kind() == SyntaxKind.AwaitKeyword && localDeclaration.UsingKeyword.Kind() == SyntaxKind.UsingKeyword)
+                    if (localDeclaration.UsingKeyword.Kind() == SyntaxKind.UsingKeyword)
```

Comment:
i preserved this logic.  But i dont' actually think it's necessary.  this basically ensures that we only highlight `await` in `await using var whatever` and not in `await var whatever`.  However, the latter is broken code, so it would be fine IMO to highlight there.